components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/AccountConfirmationValidationHandler.java

```diff
@@ -126,6 +126,12 @@ public void handleEvent(Event event) throws IdentityEventException {
                 IdentityUtil.setIdentityErrorMsg(customErrorMessageContext);
                 throw new IdentityEventException(IdentityCoreConstants.USER_EMAIL_NOT_VERIFIED_ERROR_CODE,
                         "User : " + Utils.maskIfRequired(userName) + "'s email is not confirmed yet.");
+            } else if (!isSelfSignupEnabled && operationStatus && !isUserEmailVerificationScenario(user)) {
```

Comment:
```suggestion
            } else if (!isSelfSignupEnabled && operationStatus && isUserEmailVerificationScenario(user)) {
```

Should we need not operator?